examples/jsm/utils/GeometryCompressionUtils.js

```diff
@@ -0,0 +1,812 @@
+                let tmpy = Math.round((y + 1) * 0.5 * 65535);
+                let tmpz = Math.round((z + 1) * 0.5 * 65535);
+                return new Uint16Array([tmpx, tmpy, tmpz]);
+            } else {
+                console.error("number of bytes must be 1 or 2");
+            }
+        },
+
+        defaultDecode: function (array, bytes) {
+            if (bytes == 1) {
+                return [
+                    ((array[0] / 255) * 2.0) - 1.0,
+                    ((array[1] / 255) * 2.0) - 1.0,
+                    ((array[2] / 255) * 2.0) - 1.0,
+                ]
+            } else if (bytes == 2) {
+                return [
+                    ((array[0] / 65535) * 2.0) - 1.0,
+                    ((array[1] / 65535) * 2.0) - 1.0,
+                    ((array[2] / 65535) * 2.0) - 1.0,
+                ]
+            } else {
+                console.error("number of bytes must be 1 or 2");
+            }
+
+        },
+
+        // for `Angles` encoding
+        anglesEncode: function (x, y, z) {
+            let normal0 = parseInt(0.5 * (1.0 + Math.atan2(y, x) / Math.PI) * 65535);
+            let normal1 = parseInt(0.5 * (1.0 + z) * 65535);
+            return new Uint16Array([normal0, normal1]);
+        },
+
+        // for `OCT` encoding
+        octEncodeBest: function (x, y, z, bytes) {
+            var oct, dec, best, currentCos, bestCos;
+
+            // Test various combinations of ceil and floor
+            // to minimize rounding errors
+            best = oct = octEncodeVec3(x, y, z, "floor", "floor");
+            dec = octDecodeVec2(oct);
+            currentCos = bestCos = dot(x, y, z, dec);
+
+            oct = octEncodeVec3(x, y, z, "ceil", "floor");
+            dec = octDecodeVec2(oct);
+            currentCos = dot(x, y, z, dec);
+
+            if (currentCos > bestCos) {
+                best = oct;
+                bestCos = currentCos;
+            }
+
+            oct = octEncodeVec3(x, y, z, "floor", "ceil");
+            dec = octDecodeVec2(oct);
+            currentCos = dot(x, y, z, dec);
+
+            if (currentCos > bestCos) {
+                best = oct;
+                bestCos = currentCos;
+            }
+
+            oct = octEncodeVec3(x, y, z, "ceil", "ceil");
+            dec = octDecodeVec2(oct);
+            currentCos = dot(x, y, z, dec);
+
+            if (currentCos > bestCos) {
+                best = oct;
+                bestCos = currentCos;
+            }
+
+            return best;
+
+            function octEncodeVec3(x0, y0, z0, xfunc, yfunc) {
+                var x = x0 / (Math.abs(x0) + Math.abs(y0) + Math.abs(z0));
+                var y = y0 / (Math.abs(x0) + Math.abs(y0) + Math.abs(z0));
+
+                if (z < 0) {
+                    var tempx = x;
+                    var tempy = y;
+                    tempx = (1 - Math.abs(y)) * (x >= 0 ? 1 : -1);
+                    tempy = (1 - Math.abs(x)) * (y >= 0 ? 1 : -1);
```

Comment:
Another lgtm issue: `tempx` and `tempy` are always assigned values here, so you can write the code like so:
```Js
var tempx = (1 - Math.abs(y)) * (x >= 0 ? 1 : -1);
var  tempy = (1 - Math.abs(x)) * (y >= 0 ? 1 : -1);
```
and delete `var tempx = x;` and `var tempy = y;`.